### PR TITLE
2PC PRF aio

### PR DESCRIPTION
**Pull request by sinui0**
This PR implements our 2PC PRF protocol in tls-2pc-aio. It also touches up the core crate, mostly renaming and slightly changing up the interface (should be no change to business logic). Additionally, I added lazy loading for the circuits and benches.

### CODE REVIEW

**Review by th4s — COMMENTED** [review]
Nice work!

File: tls-2pc-aio/src/prf/circuits/c1.rs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                // convert to little-endian
+                .rev()
+                .copied()
+                .collect::<Vec<u8>>(),
+        )
+        .expect("P256SecretShare should always be 32 bytes");
+
+    let mask: Vec<u8> = thread_rng().gen::<[u8; 32]>().to_vec();
+    let input_mask = circ
+        .input(2)
```
[the remark this answers is not in the document]
Comment (th4s) [reply tls-2pc-aio/src/prf/circuits/c1.rs:33]: Sounds good :)

File: tls-2pc-aio/src/prf/mod.rs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            .map(|(a, b)| a ^ b)
+            .collect::<Vec<u8>>();
+        let siv = leader_keys
+            .siv()
+            .iter()
+            .zip(follower_keys.siv())
+            .map(|(a, b)| a ^ b)
+            .collect::<Vec<u8>>();
+
+        let (expected_swk, expected_cwk, expected_siv, expected_civ) =
```
Comment (themighty1) [inline tls-2pc-aio/src/prf/mod.rs:173]: There must be an error somewhere because the correct return order is cwk, swk, civ, siv

**Review by themighty1 — CHANGES_REQUESTED** [review]
gw

**Pull request merged**